### PR TITLE
Lua script input_event handling and update documentation 

### LINKED ISSUE #885: Developer Manual is not up-to-date Lua 5.4 and Ubuntu 21 are required
XCSoar versions having and not having the problem
-------------------------------------------------
master branch:
The developer Manual doesn't tell anything about  which Ubuntu version to use
it refers to Lua 5.2 as a depenency. This won't work.
Another dependency is G++10.

I used developer Manual 7.0 to build XCSoar from github

Expected behavior
-----------------

Actual behavior
---------------

Steps to reproduce the behavior
-------------------------------
Try an older Ubuntu. You will fail because G++10 is required.
the other messages will follow during build process.

Do you have any idea what may have caused this?
-----------------------------------------------

Do you have an idea how to solve the issue?
-------------------------------------------
Isn't it better to list the known depenencies in the github start page(https://github.com/XCSoar/XCSoar)?
This is easier to maintain.

### PR DESCRIPTION
**Pull request by volkmarnissen**
Fix bug in lua/InputEvent.cpp:
A lowercase key was not recognized by the Lua input_event.
Added ToUpperASCII in the Lua::FireKey method

Improved documentation in build.rst: Added reference to ubuntu 21 rsp.
debian 11

Added missing documentation in lua.rst.
input_event is now documented.
Input/InputEvents.cpp: If a lua function is configured in init.lua,
the event won't be passed to depreceated InputEvent handling anymore.

Brief summary of the changes
----------------------------

Related issues and discussions
------------------------------

Closes #885

### CODE REVIEW

**Review by lordfolken — CHANGES_REQUESTED** [review]
Please make your commit message more descriptive.

**Comment by volkmarnissen** [comment]
Added more detailed information about all commits to the last commit message.
It also should solve the compile issue, which didn't occure locally.

**Review by lordfolken — CHANGES_REQUESTED** [review]
This is a full path, and shouldn't be here.

File: .gitignore [hunk 1]
```diff
@@ -35,3 +35,4 @@ build/local-config.mk
 .*.swp
 
 .idea
+/sysroots/
```
Comment (lordfolken) [inline .gitignore:38]: full path. Shouldn't be a fully qualified path.

**Review by lordfolken — CHANGES_REQUESTED** [review]

**Comment by MaxKellermann** [comment]
A PR must not contain reverts or merge commits.

**Comment by MaxKellermann** [comment]
Reset your PR branch to origin/master and cherry-pick all the commits you want to have in there, then force-push.
When I work on git branches, stgit is a valuable tool (I can't work without it). It helps you avoid the troubles you had.

**Comment by volkmarnissen** [comment]
The willl be reopened with cleaned commits.

**Pull request closed without merging**